src/core/server/http/logging/get_response_log.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import querystring from 'querystring';
+import { isBoom } from '@hapi/boom';
+import type { Request } from '@hapi/hapi';
+import numeral from '@elastic/numeral';
+import { LogMeta } from '@kbn/logging';
+import { EcsEvent } from './ecs';
+import { getResponsePayloadBytes } from './get_payload_size';
+
+/**
+ * Converts a hapi `Request` into ECS-compliant `LogMeta` for logging.
+ *
+ * @internal
+ */
+export function getEcsResponseLog(request: Request): LogMeta {
+  const { path, response } = request;
+  const method = request.method.toUpperCase();
+
+  const query = querystring.stringify(request.query);
+  const pathWithQuery = query.length > 0 ? `${path}?${query}` : path;
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  const status_code = isBoom(response) ? response.output.statusCode : response.statusCode;
+
+  // borrowed from the hapi/good implementation
+  const responseTime = (request.info.completed || request.info.responded) - request.info.received;
+  const responseTimeMsg = !isNaN(responseTime) ? ` ${responseTime}ms` : '';
+
+  const bytes = getResponsePayloadBytes(response);
+  const bytesMsg = bytes ? ` - ${numeral(bytes).format('0.0b')}` : '';
+
+  const meta: EcsEvent = {
+    ecs: { version: '1.7.0' },
```

Comment:
nit: I would extract that to a constant, even if only in non-exported from the current file for now?